### PR TITLE
docs: sponsors update

### PR DESCRIPTION
**Pull request by JacobCoffee**
Add sponsors section to our readme and docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,36 @@ app = Litestar(route_handlers=[hello_world])
...
+Litestar is an open-source project, and we enjoy the support of our sponsors to help fund the exciting
+work we do.
+
+If you would like to support the work that we do please consider [becoming a sponsor][sponsor-github]
+on [GitHub][sponsor-github] or [Open Collective][sponsor-oc].
+
+We also participate in pledge-based sponsorship with [Polar][sponsor-polar].
+
+Does your company require third-party services offer support contracts?
+We also provide SLAs and support contracts for Litestar.
```
Comment (guacs) [inline README.md:127]: Should this be included? In our Github Sponsors, we explicitly state that there is no SLA. Maybe this wording could be changed until we reach a decision on #2861?

Comment (JacobCoffee) [reply README.md:127]: Ahh yeah good point. I don't want to hold up the sponsor badges going in as its been promised so I can take it out for now.

File: README.md [hunk 2]
```diff
@@ -113,6 +113,36 @@ app = Litestar(route_handlers=[hello_world])
...
+If you are interested in this, please check out our tiers on [GitHub Sponsor][sponsor-github]
+, [Open Collective][sponsor-oc], or (Coming Soon) [Polar][sponsor-polar].
```
Comment (JacobCoffee) [inline README.md:129]:
```suggestion
```

File: docs/index.rst [hunk 3]
```diff
@@ -132,6 +132,44 @@ and
...
+Support Agreements
+~~~~~~~~~~~~~~~~~~
+
+For businesses that mandate or have regulatory requirements of support contracts for open-source solutions,
+Litestar offers solutions to meet these requirements.
+
+Our commitment includes comprehensive support, ensuring that your organization receives timely and effective assistance.
+Explore our various support tiers on `GitHub <sponsor-github_>`_,
+`Open Collective <sponsor-oc_>`_, and soon on `Polar <sponsor-polar_>`_,
+to secure a partnership that aligns with your business needs.
```
Comment (JacobCoffee) [inline docs/index.rst:156]:
```suggestion
```

**Comment by JacobCoffee** [comment]
@chris-telemetry could you confirm this is okay? Using the official `svg` and brand color from the website but you may want another image that would fill out the space better (like the T/S square logo?)
@marclave could you confirm this looks okay as well?

File: README.md [hunk 4]
```diff
@@ -113,6 +113,32 @@ app = Litestar(route_handlers=[hello_world])
...
+[//]: # "Note to maintainers: Highest sponsors first; no more than 3 per row - create new div if needed"
+
+<div style="display: flex; justify-content: center; align-items: center; gap: 8px;">
+  <a href="https://scalar.com">
+    <img src="https://raw.githubusercontent.com/litestar-org/branding/main/assets/sponsors/scalar.svg" alt="Scalar.com" style="border-radius: 10px; width: auto; max-height: 150px;"/> <!-- Adjust max-height as needed -->
+  </a>
+  <a href="https://telemetrysports.com/">
+    <img src="https://raw.githubusercontent.com/litestar-org/branding/main/assets/sponsors/telemetry-sports-temp.svg" alt="Telemetry Sports" style="border-radius: 10px; width: auto; max-height: 150px;"/> <!-- Adjust max-height as needed -->
+  </a>
+</div>
```
Comment (provinzkraut) [inline README.md:136]: Can't we autogenerate this section? I can see this getting outdated fairly quickly otherwise.

**Comment by marclave** [comment]
> @chris-telemetry could you confirm this is okay? Using the official `svg` and brand color from the website but you may want another image that would fill out the space better (like the T/S square logo?) @marclave could you confirm this looks okay as well?

@JacobCoffee this is awesome!

**Review by cofin — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (JacobCoffee) [reply README.md]: Hopefully we can do this via #2866 soon.

**Comment by JacobCoffee** [comment]
Unfortunately my markdown editor != GitHubs end-game styling..

image

not sure what happened. looking..

**Comment by chris-telemetry** [comment]
Not sure if you've tried VS Code's GitHub markdown extension but it's normally spot on. Could help!

**Pull request merged**